sql/src/main/java/io/crate/expression/tablefunctions/ValuesFunction.java

```diff
@@ -0,0 +1,185 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.tablefunctions;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.WhereClause;
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Reference;
+import io.crate.metadata.ReferenceIdent;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Routing;
+import io.crate.metadata.RoutingProvider;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.table.StaticTableInfo;
+import io.crate.metadata.table.TableInfo;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.ObjectType;
+import org.elasticsearch.cluster.ClusterState;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static io.crate.metadata.functions.params.Param.ANY_ARRAY;
+import static io.crate.types.DataTypes.isArray;
+
+public class ValuesFunction {
+
+    public static final String NAME = "values";
+    private static final RelationName TABLE_IDENT = new RelationName("", NAME);
+
+    private static class ValuesTableFunctionImplementation extends TableFunctionImplementation<List<Object>> {
+
+        private final FunctionInfo info;
+
+        private ValuesTableFunctionImplementation(FunctionInfo info) {
+            this.info = info;
+        }
+
+        @Override
+        public FunctionInfo info() {
+            return info;
+        }
+
+        @Override
+        public final Iterable<Row> evaluate(TransactionContext txnCtx,
+                                            Input<List<Object>>[] arguments) {
+            int numCols = arguments.length;
+            Object[] cells = new Object[numCols];
+            RowN row = new RowN(cells);
+            Iterator<Object>[] iteratorsPerColumn = iteratorsFrom(arguments);
+
+            return () -> new Iterator<>() {
```

Comment:
We could probably share the `Iterable` `Iterator` part with `unnest` by making the `iteratorsPerColumn` a parameter.